### PR TITLE
Fix Settings import

### LINKED ISSUE #7075: "Search Suggestions" setting not imported
### Checklist

- [x] I am using the latest version - curent dev branch and soon to be 0.21.10 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
1. Go to Settings > Content > Search Suggestions and uncheck "remote xyz"
2. Go to Settings > Content > Export database and export it
3. Go to Settings > Content > Search Suggestions and check "remote xyz"
4. Import the database from step 2
5.  Go to Settings > Content > Search Suggestions

### Actual behavior

"remote xyz" is checked

### Expected behavior

"remote xyz" is unchecked

### Device info
not relevant

### Note
It took me more time writing this issue, than fixing it myself. So I would appreciate if it was actually left for newcomers

### PR DESCRIPTION
**Pull request by laksh-21**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- The "Search Suggestions" setting in Settings > Content > Search Suggestions is a `MultiSelectListPreference` and that exports as a `HashSet` of all the enabled options.
- The export was going through properly, but the import function did not have a case to handle the value if it was a `HashSet` in the key-value pair.
- Since the incoming value was not being processed, the setting was just reverting to its default values.
- I have added the case to handle when the value is a `HashSet`. 
- The `Suppress` annotation was added because value is of type `any?` and it needs to be converted to `MutableSet<String>?` which throws an `Unchecked Cast` warning. As far as I know, this warning can be suppressed without any problems, because the `HashSet` will always be convertible to `MutableSet<String>`. 

#### Fixes the following issue(s)

- Fixes #7075

#### APK testing 

APK for the app: [debug_app.apk](https://github.com/TeamNewPipe/NewPipe/suites/3870923255/artifacts/96326695)
**Edit:** added the APK

Also, this is my first time contributing :). Can I get a feedback on how well this PR description was written?

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by opusforlife2** [comment]
Hi, @laksh-21, and thank you for contributing to Newpipe! I think your PR description was well written. I made some small edits to make it look better, which you can see in detail using the edit history.

**Comment by laksh-21** [comment]
@opusforlife2 oh thanks! I'll keep that in mind next time. Didn't realize I made so many types T_T.
By the way, does the review usually take a long time?

**Comment by opusforlife2** [comment]
@laksh-21 That completely depends on developer availability. Since this is a completely voluntary project, that could mean hours/days/weeks or even months. But usually small PRs like yours get reviewed fairly quickly.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt [hunk 1]
```diff
@@ -91,6 +91,10 @@ class ContentSettingsManager(private val fileLocator: NewPipeFileLocator) {
                         is String -> {
                             preferenceEditor.putString(key, value)
                         }
+                        is HashSet<*> -> {
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt:94]:
```suggestion
                        is Set<*> -> {
```
Set exactly matches the type of ``putStringSet`` - so let's use it.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt [hunk 2]
```diff
@@ -91,6 +91,10 @@ class ContentSettingsManager(private val fileLocator: NewPipeFileLocator) {
...
+                            @Suppress("UNCHECKED_CAST")
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt:95]:
```suggestion
                            // There are currently only Sets with type String possible
                            @Suppress("UNCHECKED_CAST")
```

Adding a comment here why the warning is ignored

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt [hunk 3]
```diff
@@ -91,6 +91,10 @@ class ContentSettingsManager(private val fileLocator: NewPipeFileLocator) {
...
+                            preferenceEditor.putStringSet(key, value as MutableSet<String>?)
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt:96]:
```suggestion
                            preferenceEditor.putStringSet(key, value as Set<String>?)
```
Set exactly matches the type of ``putStringSet`` - so let's use it.

**Review by litetex — CHANGES_REQUESTED** [review]
Code generally looks good to me (LGTM)

However I request some minor changes.

Tested this (with my minor changes) and works as expected ✔️

**Comment by laksh-21** [comment]
Hi @litetex I've made the requested changes :)

**Review by litetex — APPROVED** [review]
LGTM

**Pull request merged**